src/patch.jl

```diff
@@ -126,19 +126,26 @@ end
 ```
 """
 function apply(body::Function, pe::PatchEnv)
-    prev_pe = get_active_env()
-    set_active_env(merge(prev_pe, pe))
-    try
-        return body()
-    finally
-        set_active_env(prev_pe)
-    end
+    merged_pe = merge(PATCH_ENV[], pe)
+    return with_active_env(body, merged_pe)
 end
 
 function apply(body::Function, patches; debug::Bool=false)
     return apply(body, PatchEnv(patches, debug))
 end
 
-const PATCH_ENV = Ref{PatchEnv}(PatchEnv())
-set_active_env(pe::PatchEnv) = (PATCH_ENV[] = pe)
-get_active_env() = PATCH_ENV[]
+if VERSION > v"1.11-"
+    const PATCH_ENV = ScopedValue(PatchEnv())
+    with_active_env(body::Function, pe::PatchEnv) = with(body, PATCH_ENV => pe)
+else
+    function with_active_env(body::Function, pe::PatchEnv)
+        old_pe = get_active_env
+        try
+            PATCH_ENV[] = pe
+            body()
+        finally
+            PATCH_ENV[] = old_pe
+        end
+    end
```

Comment:
Needs work

Reply:
Specifically:
- `get_active_env` isn't actually called
- I'd move `merge` back into this function as now if this function is called directly the passed in `pe` is not combined with the `old_pe`.

Reply:
re: merge: isn't that the difference betweem this function and `apply` ?
`apply` merges current active enviroment then calls this.

If I were to combine them i would still end up with a very short helper for this functionality.

Maybe I sould rename it with a leading underscope to make it clear it shouldn;t be called directly?